packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -498,7 +498,8 @@ class _NavigationRailState extends State<NavigationRail> with TickerProviderStat
                           labelTextStyle: widget.selectedIndex == i ? selectedLabelTextStyle : unselectedLabelTextStyle,
                           padding: widget.destinations[i].padding,
                           onTap: () {
-                            widget.onDestinationSelected!(i);
+                            if(widget.onDestinationSelected != null)
+                              widget.onDestinationSelected!(i);
```

Comment:
Just a style choice, however, I feel this is more concise:

```suggestion
                            widget.onDestinationSelected?.call(i);
```

Reply:
I was going to do that, but the BottomNavigationBar uses that style which made me opt in
```dart
 onTap: () {
    if (widget.onTap != null)
      widget.onTap!(i);
},
```